crates/node/src/providers/robust_ecdsa.rs

```diff
@@ -0,0 +1,251 @@
+            channel,
+        )
+        .await
+    }
+
+    async fn process_channel(&self, channel: NetworkTaskChannel) -> anyhow::Result<()> {
+        match channel.task_id() {
+            MpcTaskId::RobustEcdsaTaskId(task) => match task {
+                RobustEcdsaTaskId::KeyGeneration { .. } => {
+                    anyhow::bail!("Key generation rejected in normal node operation");
+                }
+                RobustEcdsaTaskId::KeyResharing { .. } => {
+                    anyhow::bail!("Key resharing rejected in normal node operation");
+                }
+                RobustEcdsaTaskId::Presignature { id, domain_id } => {
+                    self.run_presignature_generation_follower(channel, id, domain_id)
+                        .await?;
+                }
+                RobustEcdsaTaskId::Signature {
+                    id,
+                    presignature_id,
+                } => {
+                    self.make_signature_follower(channel, id, presignature_id)
+                        .await?;
+                }
+            },
+
+            _ => anyhow::bail!(
+                "eddsa task handler: received unexpected task id: {:?}",
+                channel.task_id()
+            ),
+        }
+        Ok(())
+    }
+
+    async fn spawn_background_tasks(self: Arc<Self>) -> anyhow::Result<()> {
+        let generate_presignatures = self
+            .per_domain_data
+            .iter()
+            .map(|(domain_id, data)| {
+                tracking::spawn(
+                    &format!("generate presignatures for domain {}", domain_id.0),
+                    Self::run_background_presignature_generation(
+                        self.client.clone(),
+                        self.mpc_config.clone(),
+                        self.config.presignature.clone().into(),
+                        *domain_id,
+                        data.presignature_store.clone(),
+                        data.keyshare.clone(),
+                    ),
+                )
+            })
+            .collect::<Vec<_>>();
+
+        for task in generate_presignatures {
+            task.await??;
+        }
+
+        Ok(())
+    }
+}
+
+pub(super) fn get_number_of_signers(threshold: usize, _number_of_participants: usize) -> usize {
+    // TODO: this is the case for the other schemes, might not be our best choice for this one
+    threshold
+}
+
+/// This function translates the current threshold from the contract
+/// to the threshold expected by the robust-ecdsa scheme, which
+/// is semantically different.
+/// The function should be no longer needed when these issues are solved:
+/// https://github.com/near/threshold-signatures/issues/255
+/// https://github.com/near/mpc/issues/1649
+pub(super) fn translate_threshold(threshold: usize, number_of_participants: usize) -> usize {
+    let number_of_signers = get_number_of_signers(threshold, number_of_participants);
+    (number_of_signers - 1) / 2
+}
```

Comment:
We should add some unit tests for this function. This should be trivial but it would illustrate very explicitly how the values translate.